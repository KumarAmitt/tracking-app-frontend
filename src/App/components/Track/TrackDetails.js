import React, { useEffect } from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import AppBar from '../AppBar/AppBar';
import { getProducts, loadProducts } from '../../../store/slicers/product';
import './style/TrackDetails.css';
import Stats from './Stats';
import { getUserInfo, loadSession } from '../../../store/slicers/user';

const TrackDetails = ({ location }) => {
  const { info, date } = location;
  const title = 'Track Details';
  const dispatch = useDispatch();
  const userInfo = useSelector(getUserInfo);
  const products = useSelector(getProducts);
  const todayTotal = info ? info.map((e) => e.premium).reduce((a, b) => a + b) : 0;

  useEffect(() => {
    dispatch(loadSession());
    dispatch(loadProducts());
  }, []);

  if (!userInfo.logged_in) {
    return <Redirect to="/track" />;
  }

  return (
    <>
      <AppBar title={title} link="/track" />
      <div className="details">
        <div className="details-header">
          <div className="date">{date}</div>
          <Stats today={todayTotal || 0} />
        </div>
        <div className="details-items">
          {
            info && info.map((e) => (
              <div key={e.id} className="details-item">
                <div className="ins-type">{products.length > 0 ? products.filter((f) => f.id === e.product_id)[0].product_name : ''}</div>
                <div className="premium">
                  <span>&#8377;</span>
                  {' '}
                  {e.premium}
                </div>
                <div className="appl-id">{e.application_id}</div>
              </div>
            ))
          }
        </div>
      </div>
    </>
  );
};

TrackDetails.propTypes = {
  location: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default TrackDetails;
