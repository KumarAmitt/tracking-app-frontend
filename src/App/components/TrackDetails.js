import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import AppBar from './AppBar';
import { getProducts, loadProducts } from '../../store/slicers/product';

const TrackDetails = ({ location }) => {
  const { info, date } = location;
  const title = 'Track Details';
  const dispatch = useDispatch();
  const products = useSelector(getProducts);
  const todayTotal = info ? info.map((e) => e.premium).reduce((a, b) => a + b) : 0;

  useEffect(() => {
    dispatch(loadProducts());
  }, []);

  console.log(todayTotal);
  return (
    <>
      <AppBar title={title} />
      <Link to="/track">Go Back</Link>
      <div>{date}</div>
      <hr />
      {
        info && info.map((e) => (
          <div key={e.id}>
            <div>{products.length > 0 ? products.filter((f) => f.id === e.product_id)[0].product_name : ''}</div>
            <div>{e.premium}</div>
            <div>{e.application_id}</div>
            <hr />
          </div>
        ))
      }
    </>
  );
};

TrackDetails.propTypes = {
  location: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default TrackDetails;
