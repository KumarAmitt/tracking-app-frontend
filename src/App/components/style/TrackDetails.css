.details {
  height: calc(100vh - 7em);
  margin-top: 3em;
}

.details-header {
  padding: 0.5em;
}

.details-header .date {
  text-align: center;
}

.details-items {
  background-color: #eee;
  padding: 1em 0.8em;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-auto-rows: minmax(min-content, max-content);
  grid-gap: 0.8em;
  height: calc(100vh - 13.3em);
  overflow: auto;
}

.details-item {
  background-color: #fff;
  padding: 0.4em;
  text-align: center;
}

.details-item * + * {
  padding-top: 0.4em;
}

.ins-type {
  font-size: 0.8rem;
}

.appl-id {
  font-size: 0.9rem;
}

.premium {
  font-weight: bold;
  color: var(--accent-primary);
}
