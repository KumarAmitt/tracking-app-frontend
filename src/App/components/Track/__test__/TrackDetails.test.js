import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureAppStore from '../../../../store/configureStore';
import TrackDetails from '../TrackDetails';

const store = configureAppStore();

describe('TrackDetails', () => {
  it('should match the snapshot of TrackDetails component', () => {
    const info = [{ id: 1, product_id: 1, premium: 2000 }];
    const date = '2021-08-29';
    const location = { info, date };
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <TrackDetails location={location} />
        </BrowserRouter>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
