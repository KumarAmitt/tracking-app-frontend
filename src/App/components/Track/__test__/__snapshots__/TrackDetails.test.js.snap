// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackDetails should match the snapshot of TrackDetails component 1`] = `
Array [
  <div
    className="appbar"
  >
    <a
      href="/track"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
        />
      </svg>
    </a>
    <div
      className="title"
    >
      Track Details
    </div>
  </div>,
  <div
    className="details"
  >
    <div
      className="details-header"
    >
      <div
        className="date"
      >
        2021-08-29
      </div>
      <div
        className="stats"
      >
        <div
          className="stats-item"
        >
          <div>
            2000
          </div>
          <div>
            Day
          </div>
        </div>
        <div
          className="stats-item"
        >
          <div>
            0
          </div>
          <div>
            Achieved
          </div>
        </div>
        <div
          className="stats-item"
        >
          <div>
            700000
          </div>
          <div>
            Lag
          </div>
        </div>
        <div
          className="stats-item"
        >
          <div>
            700000
          </div>
          <div>
            Target
          </div>
        </div>
      </div>
    </div>
    <div
      className="details-items"
    >
      <div
        className="details-item"
      >
        <div
          className="ins-type"
        >
          
        </div>
        <div
          className="premium"
        >
          <span>
            â‚¹
          </span>
           
          2000
        </div>
        <div
          className="appl-id"
        />
      </div>
    </div>
  </div>,
]
`;
